bin/03_impute_infection_dates.R

```diff
@@ -0,0 +1,78 @@
+library(dplyr)
+library(EpiGenR)
+library(magrittr)
+library(readr)
+library(phytools)
+library(lubridate)
+
+args <- commandArgs(TRUE)
+timetree <- args[1] # ex: "~/code/nf-core-epigen/results/newicks/tree_california.nwk"
+tip_dates_file <- args[2] # ex: "~/code/nf-core-epigen/results/gisaid/metadata.tsv"
+cleaned_timeseries <- file.path(args[3]) # ex: "~/code/nf-core-epigen/results/timeseries_cleaned/summary_california_timeseries_new_cases.tsv"
+
+
+# reformat labels on trees to include date of sampling
+# -----------------------------------------------------
+trs <- mapply(function (x, y) {
+  tr <- read.newick(x)
+  node_dates <- read_tsv(
+    y, comment="#", col_names=c(
+      "node", "date", "date_median", "date_lower", "date_upper"
+      )) %>% filter(as.character(date)!="--")
+  
+  tr <- keep.tip(tr, which(tr$tip.label %in% node_dates$node))
+  
+  tr$tip.label %<>% paste(node_dates$date[match(., node_dates$node)], sep="_")
```

Comment:
lines 18-25 should be updated to use the metadata.tsv table.
Suggested change:
```suggestion
  node_dates <- read_tsv(y)
  tr <- keep.tip(tr, which(tr$tip.label %in% node_dates$gisaid_epi_isl))
  tr$tip.label %<>% paste(node_dates$date[match(., node_dates$gisaid_epi_isl)], sep="_")
```

Reply:
oh perfect, i totally forgot about our earlier conversation about `gisaid_epi_isl`